pkg/cableengine/healthchecker/pinger.go

```diff
@@ -111,9 +111,17 @@ func (p *pingerInfo) doPing() error {
 		if pinger.PacketsSent-pinger.PacketsRecv > int(p.maxPacketLossCount) {
 			p.Lock()
 			defer p.Unlock()
+			// The counter will be reset to once to ensure that it is continuous packet loss
+			// before marking the connection as down.
+			if pinger.PacketsSent <= int(p.maxPacketLossCount)+1 {
+				p.failureMsg = fmt.Sprintf("Failed to successfully ping the remote endpoint IP %q", p.ip)
+				p.connectionStatus = ConnectionError
+
+				pinger.Stop()
```

Comment:
But we stop the pinger forever? :?

I'm not understand the patch :)

Reply:
I mean, when is it started again?

Reply:
We restart the pinger in infinite [1] for loop and will be stopped only if a stop signal, which happens when the endpoint is deleted.

[1] https://github.com/submariner-io/submariner/blob/master/pkg/cableengine/healthchecker/pinger.go#L67

Reply:
I don't understand how's it started again, I'm missing something,

Reply:
ahh, it's pinger.stop , which you created in L89